packages/gui/src/components/nfts/gallery/NFTGallery.tsx

```diff
@@ -253,6 +254,21 @@ export default function NFTGallery() {
     };
   }, []);
 
+  React.useEffect(() => {
+    if (allowNFTsFiltered.length) {
+      if (document.getElementById('scroll-helper') && scrollPosition.get('position') > 0) {
+        setTimeout(() => {
+          if (document.getElementById('scroll-helper')?.parentNode) {
+            (document.getElementById('scroll-helper')?.parentNode as HTMLElement).scrollTo(
+              0,
+              scrollPosition.get('position')
+            );
+          }
+        }, 0);
+      }
```

Comment:
A couple of suggestions:

1) `document.getElementById('scroll-helper')` is called up to 3 times. Can we just assign the result to a local const? ```const scrollHelper = document.getElementById('scroll-helper');```

2) I'm finding it harder and harder to read the code with so many useEffects and other hooks that have inlined functions. To improve the code clarity, I would like to see a couple of functions named `setScrollPosition()` and `restoreScrollPosition()`. Can we refactor this code to support that?